dix/req.c

```diff
@@ -2214,8 +2263,17 @@ static int dix_cas_rops_alloc(struct m0_dix_req *req)
 		}
 		for (j = 0; j < rec_op->dgp_units_nr; j++) {
 			unit = &rec_op->dgp_units[j];
-			if (dix_pg_unit_skip(req, unit))
+			if (dix_pg_unit_skip(req, unit)) {
+				if (dtx != NULL &&
```

Comment:
Do we really need to check it everywhere in this function? Cannot we just check it once in the very beginning?

Reply:
For example in this code, the loop, it seems, does not do anything if `dtx == NULL`, so why to check for it inside each iteration instead of checking for it once outside the loop?

Reply:
```
Is there anything in this function that should be done when dtx == NULL?
```
Yes, sure, this is a common code for read/modify operations. For read operations (DIX GET) we don't have dtx.

Reply:
> For example in this code, the loop, it seems, does not do anything if dtx == NULL, so why to check for it inside each iteration instead of checking for it once outside the loop?

No, in opposite, by condition dtx != NULL we are doing a small part to fill skipped_sdevs array, all other logic (allocation of CAS requests) is done despite dtx is NULL or not.

Reply:
Ok, thanks.